### PR TITLE
Use where syntax for SharedArray constructor

### PR DESCRIPTION
**Pull request by musm**

### CODE REVIEW

File: base/sharedarray.jl [hunk 1]
```diff
@@ -147,7 +147,7 @@ SharedArray{T}(m::Integer, n::Integer; kwargs...) where {T} =
 SharedArray{T}(m::Integer, n::Integer, o::Integer; kwargs...) where {T} =
     SharedArray{T,3}(m, n, o; kwargs...)
 
-function (::Type{SharedArray{T,N}})(filename::AbstractString, dims::NTuple{N,Int}, offset::Integer=0;
+function SharedArray{T,N}(filename::AbstractString, dims::NTuple{N,Int}, offset::Integer=0;
                                     mode=nothing, init=false, pids::Vector{Int}=Int[]) where T where N
```
Comment (JeffBezanson) [inline base/sharedarray.jl]: Indent should be fixed here.

Comment (musm) [inline base/sharedarray.jl]: thanks, fixed

File: base/sharedarray.jl [hunk 2]
```diff
@@ -147,8 +147,8 @@ SharedArray{T}(m::Integer, n::Integer; kwargs...) where {T} =
 SharedArray{T}(m::Integer, n::Integer, o::Integer; kwargs...) where {T} =
     SharedArray{T,3}(m, n, o; kwargs...)
 
-function (::Type{SharedArray{T,N}})(filename::AbstractString, dims::NTuple{N,Int}, offset::Integer=0;
-                                    mode=nothing, init=false, pids::Vector{Int}=Int[]) where T where N
+function SharedArray{T,N}(filename::AbstractString, dims::NTuple{N,Int}, offset::Integer=0;
+                          mode=nothing, init=false, pids::Vector{Int}=Int[]) where T where N
```
Comment (StefanKarpinski) [inline base/sharedarray.jl]: Can't these be combined into a single `where T, N` clause?

Comment (musm) [inline base/sharedarray.jl]: Yes they can, but the rest of the file has chosen to have them as `where T where N` so for consistency I have it like this

Comment (StefanKarpinski) [inline base/sharedarray.jl]: Ok, would probably be good to change then all to the shorter form at some point but can be done later.

**Review by Sacha0 — APPROVED** [review]
lgtm! :) (Absent objections, requests for time, or someone beating me to it, I plan to merge this tomorrow morning. Best!)

**Pull request merged**